### ISSUE TITLE
renameColumn changes collation to table default

### ISSUE THREAD

**Issue by clarkie** [sequelize/sequelize#10421]
## What are you doing?

Trying to rename a UUID column. It currently has a collation of `latin1_bin` 

```js
const tableName = 'estimate';
const columnNameBefore = 'groupId';
const columnNameAfter = 'workspaceId';

module.exports = {
  up: queryInterface => {
    return queryInterface.renameColumn(tableName, columnNameBefore, columnNameAfter, {
      charset: 'utf8',
      collate: 'latin1_bin',
    });
  },
  down: queryInterface => {
    return queryInterface.renameColumn(tableName, columnNameAfter, columnNameBefore, {
      charset: 'utf8',
      collate: 'latin1_bin',
    });
  },
};

```

## What do you expect to happen?
I want to keep the same collation on the rename

## What is actually happening?
The collation changed to the table default collation `latin1_swedish_ci`

_Output, either JSON or SQL_

```
ALTER TABLE `estimate` CHANGE `workspaceId` `groupId` CHAR(36) DEFAULT NULL;
```

__Dialect:__ mysql
__Dialect version:__ XXX
__Database version:__ 8.x?
__Sequelize version:__ ^4.39.0
__Tested with latest release:__ No

**Note :** _Your issue may be ignored OR closed by maintainers if it's not tested against latest version OR does not follow issue template._

**Comment by clarkie** [comment]
My fix was to run a query directly:

```
const tableName = '`estimate`';
const columnNameBefore = '`groupId`';
const columnNameAfter = '`workspaceId`';
const collation = '`latin1_bin';

module.exports = {
  up: queryInterface => {
    return queryInterface.sequelize.query(
      `ALTER TABLE ${tableName} CHANGE ${columnNameBefore} ${columnNameAfter} CHAR(36) DEFAULT NULL COLLATE ${collation} ;`
    );
  },
  down: queryInterface => {
    return queryInterface.sequelize.query(
      `ALTER TABLE ${tableName} CHANGE ${columnNameAfter} ${columnNameBefore} CHAR(36) DEFAULT NULL COLLATE ${collation} ;`
    );
  },
};

```

**Comment by ayavilevich** [comment]
UUID columns can't be properly renamed with renameColumn. Doing so will change their collation to a non binary one. This can cause all kinds of issues, such as problems for foreign keys.

I also had to rename the column with a manual ALTER TABLE command but I didn't want to specify the new collation manually like @clarkie to remain charset invariant. I ended up using the following fix:

```
function renameUUIDColumn(queryInterface, table, oldColumn, newColumn) {
	return queryInterface.sequelize.query(`ALTER TABLE ${table} CHANGE \`${oldColumn}\` \`${newColumn}\` CHAR(36) BINARY`);
}
```
